app/src/main/java/com/onboarding/parkingsystemkotlin/activity/ShowReservationsActivity.kt

```diff
@@ -0,0 +1,39 @@
+package com.onboarding.parkingsystemkotlin.activity
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.widget.LinearLayout
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.onboarding.parkingsystemkotlin.adapter.ReservationAdapter
+import com.onboarding.parkingsystemkotlin.database.ParkingDatabase
+import com.onboarding.parkingsystemkotlin.databinding.ActivityShowReservationsBinding
+import com.onboarding.parkingsystemkotlin.mvp.contract.ShowReservationsContract
+import com.onboarding.parkingsystemkotlin.mvp.model.ShowReservationsModel
+import com.onboarding.parkingsystemkotlin.mvp.presenter.ShowReservationsPresenter
+import com.onboarding.parkingsystemkotlin.mvp.view.ShowReservationsView
+
+class ShowReservationsActivity : AppCompatActivity() {
+    private lateinit var presenter: ShowReservationsContract.ShowReservationsPresenter
+    private lateinit var binding: ActivityShowReservationsBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityShowReservationsBinding.inflate(layoutInflater)
+        presenter = ShowReservationsPresenter(view = ShowReservationsView(this, binding), model = ShowReservationsModel(ParkingDatabase))
+        presenter.showReservations()
+        setContentView(binding.root)
+        setListeners()
+    }
+
+    private fun setListeners() {
+        binding.btnShowReservationsReturn.setOnClickListener { presenter.onReturnButtonPressed() }
```

Comment:
instead of this, you can just call the native method `onBackPressed()`

Reply:
replaced